home/utils/availability.py

```diff
@@ -0,0 +1,387 @@
+    """
+    if not users:
+        return [], 0
+
+    # Get all users' slots
+    all_user_slots = [set(get_user_slots(user)) for user in users]
+
+    # Find intersection of all users' availability
+    if not all_user_slots:
+        return [], 0
+
+    overlapping_slots = set.intersection(*all_user_slots)
+    sorted_overlap = sorted(overlapping_slots)
+
+    # Count 1-hour blocks
+    hour_blocks = count_one_hour_blocks(sorted_overlap)
+
+    return sorted_overlap, hour_blocks
+
+
+def calculate_pairwise_overlap(user1: Any, user2: Any) -> tuple[list[float], int]:
+    """
+    Find time slots where two users are available.
+
+    This is used for captain-djangonaut meetings where they need to meet 1-on-1.
+
+    Args:
+        user1: First CustomUser instance
+        user2: Second CustomUser instance
+
+    Returns:
+        Tuple of (overlapping_slots, hour_blocks_count)
+        - overlapping_slots: Sorted list of time slot values where both users overlap
+        - hour_blocks_count: Number of complete 1-hour blocks
+    """
+    user1_slots = set(get_user_slots(user1))
+    user2_slots = set(get_user_slots(user2))
+
+    overlapping_slots = user1_slots.intersection(user2_slots)
+    sorted_overlap = sorted(overlapping_slots)
+
+    hour_blocks = count_one_hour_blocks(sorted_overlap)
+
+    return sorted_overlap, hour_blocks
+
+
+def calculate_team_overlap(
+    navigator_users: list[Any],
+    captain_user: Any | None,
+    djangonaut_users: list[Any],
+) -> dict[str, Any]:
+    """
+    Calculate availability overlaps for an entire team.
+
+    Calculates:
+    1. Navigator meeting hours: All navigators + all djangonauts together
+    2. Captain 1-on-1 hours: Captain with each individual djangonaut
+
+    Args:
+        navigator_users: List of navigator CustomUser instances
+        captain_user: Captain CustomUser instance (can be None)
+        djangonaut_users: List of djangonaut CustomUser instances
+
+    Returns:
+        Dictionary with:
+        - navigator_meeting_slots: List of overlapping time slots for navigator meetings
+        - navigator_meeting_hours: Number of 1-hour blocks for navigator meetings
+        - captain_meetings: List of dicts with djangonaut info and overlap data
+        - is_valid: Boolean indicating if team meets minimum requirements (5 hours)
+    """
+    result = {
+        "navigator_meeting_slots": [],
+        "navigator_meeting_hours": 0,
+        "captain_meetings": [],
+        "is_valid": False,
+    }
+
+    # Calculate navigator meeting overlap (navigators + djangonauts, no captain)
+    navigator_meeting_participants = navigator_users + djangonaut_users
+    if navigator_meeting_participants:
+        nav_slots, nav_hours = calculate_group_overlap(navigator_meeting_participants)
+        result["navigator_meeting_slots"] = nav_slots
+        result["navigator_meeting_hours"] = nav_hours
+        result["is_valid"] = nav_hours >= 5
```

Comment:
Let's make this 5 a constant somewhere. Maybe on the Team model? There's another comment about a template that's doing a comparison on the same idea of 5 hours. Make sure these use the same constant.